lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/regex/RegexDeserializerFactory.java

```diff
@@ -30,6 +30,7 @@ public class RegexDeserializerFactory
 {
     static final String REGEX_KEY = "input.regex";
     static final String REGEX_CASE_SENSITIVE_KEY = "input.regex.case.insensitive";
+    static final String REGEX_NULL_ON_PARSE_ERROR = "input.regex.null.on.parse.error";
```

Comment:
I don't see this property in `org.apache.hadoop.hive.serde2.RegexSerDe`

Why are we adding a new property that seems to be only used in tests?

Reply:
This is to make Regex's behavior configurable when there is a parse error. By default, Regex will return a null value if it encounters a parse error. This is consistent with the previous behavior before the native Regex SerDe, but allows for users to instead have the Regex deserializer throw an error if there is an issue parsing.